Runtime/JSObject.cs

```diff
@@ -31,6 +31,11 @@ public void DefineProperties(params JSPropertyDescriptor[] descriptors)
         _value.DefineProperties(descriptors);
     }
 
+    public void DefineProperties(IReadOnlyCollection<JSPropertyDescriptor> descriptors)
```

Comment:
>IReadOnlyCollection [](http://example.com/codeflow?start=33&length=19)

I am surprised that we cannot use ReadOnlySpan. Sigh...

Reply:
Because `JSPropertyDescriptor` is a `class` it can't be a member of a `ref struct`. And it has managed objects like delegates that can't be contained by a `ref struct` either.

Reply:
Sorry, it was me ranting about the .Net limitations, not the actual change. I just wish we can use Snap in situations like this.
I guess in our code we must convert the JSPropertyDescriptor to ref struct as you did with the JSCallbackArgs.

Reply:
> I guess in our code we must convert the JSPropertyDescriptor to ref struct

I think that can be done by converting each descriptor to `napi_property_descriptor` immediately when constructed rather than as a batch later. I'm not working on it now though.